File: frontend/src/api/ai.ts
import { apiClient } from './client';

export interface ChecklistDraft {
  name: string;
  description: string;
  questions: { text: string; category: string }[];
}

export interface SuggestedAction {
  action: string;
  dueDate: string;
}

export const aiApi = {
  generateChecklist: async (topic: string, count = 8): Promise<ChecklistDraft> => {
    const { data } = await apiClient.post<ChecklistDraft>('/templates/ai-draft', { topic, count });
    return data;
  },
  suggestAction: async (findingId: string): Promise<SuggestedAction> => {
    const { data } = await apiClient.post<SuggestedAction>(`/findings/${findingId}/suggest-action`);
    return data;
  },
  ask: async (question: string): Promise<string> => {
    const { data } = await apiClient.post<{ answer: string }>('/assistant/ask', { question });
    return data.answer;
  },
};

File: backend/app/schemas/ai.py
from pydantic import BaseModel


class ChecklistGenRequest(BaseModel):
    topic: str
    count: int = 8


class ChecklistQuestionDraft(BaseModel):
    text: str
    category: str


class ChecklistDraft(BaseModel):
    name: str
    description: str
    questions: list[ChecklistQuestionDraft]


class SuggestActionOut(BaseModel):
    action: str
    dueDate: str  # ISO date (today + dueInDays)


class AssistantAsk(BaseModel):
    question: str


class AssistantAnswer(BaseModel):
    answer: str
